tests/test_views.py

```diff
@@ -77,3 +77,17 @@ def test_post_list_rendering(self):
         self.assertContains(response, post)
         self.assertContains(response, past_post)
         self.assertNotContains(response, future_post)
+
+    def test_post_detail(self):
+        """Test display of post detail view based on post's existence in db."""
+        does_not_exist_db = self.client.get(
+            reverse('post_detail', kwargs={'pk': 1}),
+        )
+        self.assertEqual(404, does_not_exist_db.status_code)
+        post = Post.objects.create(
+            author=self.user, title='test title', text='test text',
+        )
+        exist_db = self.client.get(
+            reverse('post_detail', kwargs={'pk': post.pk}),
+        )
+        self.assertEqual(200, exist_db.status_code)
```

Comment:
This is a second separate test withing the same test function. There must be just one test in a test, since it won't run if the first assertion fails and it won't show up as a distinct entry in the test report output.

Reply:
Thank you, I split one test into two, if I understood correctly.

Reply:
Marking as resolved.